### PR TITLE
AC: Add VAS head detection support

### PR DESCRIPTION
**Pull request by hojaedev**

### CODE REVIEW

**Comment by eaidova** [comment]
Jenkins ok to test

File: tools/accuracy_checker/accuracy_checker/metrics/detection.py [hunk 1]
```diff
@@ -683,22 +702,71 @@ def evaluate(self, annotations, predictions):
...
-        dt_box = list(map(lambda x: int(round(x, 0)), dt_box))
 
-        # Calculate overlapping width and height of two boxes
-        inter_width = min(gt_box[2], dt_box[2]) - max(gt_box[0], dt_box[0])
-        inter_height = min(gt_box[3], dt_box[3]) - max(gt_box[1], dt_box[1])
+class AVAHeadAccuracy(FullDatasetEvaluationMetric):
 
-        if inter_width <= 0 or inter_height <= 0:
-            return None
+    __provider__ = 'ava_head_accuracy'
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/metrics/detection.py:708]: what the difference between this metric calculation approach and youtube faces accuracy metric?

File: tools/accuracy_checker/accuracy_checker/metrics/detection.py [hunk 2]
```diff
@@ -683,22 +702,71 @@ def evaluate(self, annotations, predictions):
...
+        })
+        return parameters
+
+    def configure(self):
+        self.overlap = self.get_value_from_config('overlap')
+        self.relative_size = self.get_value_from_config('relative_size')
 
-        gt_area = (gt_box[2] - gt_box[0]) * (gt_box[3] - gt_box[1])
-        dt_area = (dt_box[2] - dt_box[0]) * (dt_box[3] - dt_box[1])
+    def submit_all(self, annotations, predictions):
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/metrics/detection.py:733]: it is standart submil_all realization, please remove this method overriding

File: tools/accuracy_checker/accuracy_checker/adapters/detection_head.py [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+import numpy as np
+
+from ..adapters import Adapter
+from ..config import NumberField, ListField, ConfigError
+from ..representation import DetectionPrediction
+
+DetectionLayerOutput = namedtuple('DetectionLayerOutput', ['prob_name', 'reg_name', 'anchor_index', 'anchor_size', 'win_scale', 'win_length', 'win_trans_x', 'win_trans_y'])
+
+class HeadDetectionAdapter(Adapter):
+    __provider__ = 'head_detection'
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/adapters/detection_head.py:27]: what the difference between this adapter and FaceDetectionAdapter?
the code looks very similar

File: tools/accuracy_checker/accuracy_checker/adapters/detection_head.py [hunk 4]
```diff
@@ -0,0 +1,159 @@
+"""
+Copyright (c) 2019 Intel Corporation
```
Comment (eizamaliev) [inline tools/accuracy_checker/accuracy_checker/adapters/detection_head.py:2]:
```suggestion
Copyright (c) 2020 Intel Corporation
```

File: tools/accuracy_checker/accuracy_checker/metrics/detection.py [hunk 5]
```diff
@@ -18,7 +18,7 @@
 import enum
```
Comment (eizamaliev) [inline tools/accuracy_checker/accuracy_checker/metrics/detection.py:18]: Please, update years to 2019-2000 in line 2.

File: tools/accuracy_checker/accuracy_checker/metrics/detection.py (hunk 1 above)
Comment (hojaedev) [reply tools/accuracy_checker/accuracy_checker/metrics/detection.py]: I've applied your feedback and reused youtube faces metric instead.

File: tools/accuracy_checker/accuracy_checker/metrics/detection.py (hunk 2 above)
Comment (hojaedev) [reply tools/accuracy_checker/accuracy_checker/metrics/detection.py]: Reflected the feedback in the new patch. Thank you.

File: tools/accuracy_checker/accuracy_checker/adapters/detection_head.py (hunk 3 above)
Comment (eaidova) [reply tools/accuracy_checker/accuracy_checker/adapters/detection_head.py:27]: @hojaedev I see that you reuse metric, it is cool, but I do not see any changes in adapter
Is there is opportunity to reuse code base of FaceDetectionAdapter (e.g. inheriting from it or unite 2 adapter into one)?

File: tools/accuracy_checker/accuracy_checker/metrics/detection.py [hunk 6]
```diff
@@ -646,42 +664,40 @@ def configure(self):
...
-                false_positive += 1
+                found = False
+                for i in range(prediction.scores.size):
+                    dt_face = [
+                        prediction.x_mins[i],
+                        prediction.y_mins[i],
+                        prediction.x_maxs[i],
+                        prediction.y_maxs[i]
+                    ]
+                    iou = self.calc_iou(gt_face, dt_face)
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/metrics/detection.py:687]: above you move iou calculation to the function, but inside the class use method.
Is there is reasons for it?
Please stay one calc_iou realization

File: tools/accuracy_checker/accuracy_checker/metrics/detection.py [hunk 7]
```diff
@@ -18,7 +18,7 @@
 import enum
 import warnings
 from typing import List
-
+from collections import defaultdict
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/metrics/detection.py:21]: I do not see this dependency usage

**Pull request merged**